### PR TITLE
DCP 509: DOI Validation

### PR DESCRIPTION
**Pull request by MightyAx**
Fixes for https://github.com/ebi-ait/dcp-ingest-central/issues/509
1. Use Europe PMC to check DOI Exists
2. If the DOI does not exist the wrangler email link includes the searched for DOI
    - Screenshot 2022-01-20 at 14 02 43
3. If the DOI is already used in a project (or more), a link to the project is given in the error message:
    - Screenshot 2022-01-20 at 14 03 33
    - Screenshot 2022-01-20 at 14 03 59
4. When Europe PMC doesn't provide all the properties we now use what we can rather than crashing.
    - Screen Recording

### CODE REVIEW

File: src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts [hunk 1]
```diff
@@ -52,31 +51,56 @@ export class AutofillProjectFormComponent implements OnInit {
...
       'operator': 'IS',
       'value': doi
     };
     query.push(criteria);
-    return this.ingestService.queryProjects(query).pipe(map(data => !!data.page.totalElements));
+    return this.ingestService.queryProjects(query).pipe(
+      map(data => !!data.page.totalElements),
+      tap(projectExists => {
+        if (projectExists) {
+          this.alertService.error('This DOI has already been used. Please contact our wranglers for further assistance', '');
```
Comment (amnonkhen) [inline src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts:79]: Have you considered pointing the user to the project that uses this doi?

File: src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts [hunk 2]
```diff
@@ -52,31 +51,56 @@ export class AutofillProjectFormComponent implements OnInit {
...
+
+  doesDoiExist(doi: string): Observable<boolean> {
+    const searchIdentifier = Identifier.DOI;
+    return this.autofillProjectService
+      .queryEuropePMC(searchIdentifier, doi)
+      .pipe(
+        map(response => response.resultList.result.length > 0),
+        tap(doiExists => {
+          if (!doiExists) {
+            this.alertService.error('This DOI cannot be found on Europe PMC. Please contact our wranglers for further assistance', '');
```
Comment (amnonkhen) [inline src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts:93]: Have you considered showing the wrangler email address here?

**Review by amnonkhen — DISMISSED** [review]
LGTM with a few comments.

File: src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts (hunk 1 above)
Comment (MightyAx) [reply src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts:79]: Great Idea. Will work on it now.

File: src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts (hunk 2 above)
Comment (MightyAx) [reply src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts:93]: Great Idea. Will work on it now.

**Review by aaclan-ebi — APPROVED** [review]

**Review by ke4 — DISMISSED** [review]

**Review by aaclan-ebi — APPROVED** [review]

**Review by ke4 — APPROVED** [review]

**Pull request merged**